tritium-caffeine/src/main/java/com/palantir/tritium/metrics/caffeine/CaffeineCacheStats.java

```diff
@@ -90,6 +97,34 @@ public static void registerCache(TaggedMetricRegistry registry, Cache<?, ?> cach
         }
     }
 
+    /**
+     * Creates a {@link StatsCounter} with tagged metrics for tracking caches hits, misses, evictions, and load times.
+     * <p>Example usage:</p>
+     * {@code
+     * Caffeine.newBuilder()
+     *     .recordStats(() -> CaffeineCacheStats.record(taggedMetricRegistry, "name")
+     *     .build()
+     * }
+     * @param registry tagged metric registry
+     * @param name safe cache name
+     * @return stats counter
+     */
+    public static StatsCounter record(TaggedMetricRegistry registry, @Safe String name) {
+        //        return CaffeineCacheTaggedMetrics.statsCounter(registry, name);
+        Function<String, MetricName> nameFunction = InternalCacheMetrics.taggedMetricName(name);
+        return new TaggedCaffeineStatsCounter(
+                registry.counter(nameFunction.apply("cache.hit.count")),
+                registry.counter(nameFunction.apply("cache.miss.count")),
+                registry.timer(nameFunction.apply("cache.load.success")),
+                registry.timer(nameFunction.apply("cache.load.failure")),
+                registry.counter(cause(nameFunction.apply("cache.eviction.count"), "total")),
+                registry.counter(cause(nameFunction.apply("cache.eviction.count"), "explicit")),
+                registry.counter(cause(nameFunction.apply("cache.eviction.count"), "replaced")),
+                registry.counter(cause(nameFunction.apply("cache.eviction.count"), "collected")),
+                registry.counter(cause(nameFunction.apply("cache.eviction.count"), "expired")),
+                registry.counter(cause(nameFunction.apply("cache.eviction.count"), "size")));
```

Comment:
Thoughts on using metric-schema for these?

Reply:
might want to do the metric-schema migration for all the Guava & Caffeine caches in a separate PR (possibly before this)

Reply:
Either way I think we'll end up with different names for metrics produced by the StatsCounter vs polling snapshots, since metric names are suffixed based on the metric type.

I think most of these counters should be meters, dropwizard counters are recorded as gauges in several backends because the values can both increment and decrement, where meters can only increase.